api/python/provisioner/srv/salt/repos/init.sls

```diff
@@ -22,18 +22,21 @@
 {% from './install.sls' import repo_added with context %}
 {% from './teardown.sls' import repo_absent with context %}
 
-{% for release, source in pillar['release']['base']['repos'].items() %}
+{% set default_install_type = 'base' %}
+{% set install_type = salt['pillar.get']('release:install_type', default_install_type) %}
```

Comment:
it means that pillar key should be set beforehand each time we call that state. E.g. the current version of SWUpgrade command doesn't do that so it will fail if that value is different than `upgrade`.

I know we discussed another option - inline pillar: it's better a bit but not a good UX as well.

I believe the best other alternative is to use different states for different routine, e.g.:
- `repos.init` for base repositories (as part of bootstrap)
- `repos.upgrade` for cases of upgrade bundles
To avoid copy paste please use [macros](https://github.com/Seagate/cortx-prvsnr/blob/pre-cortx-1.0/docs/saltstack.md#jinja-macros---sharing-the-logic-in-sls) , so that install type would be just an argument to a macros call.

Reply:
Code in `set_swupgrade_repo.py` guarantees that `install_type` pillar will be kept updated. But I implemented the new approach and uses `repos.upgrade` to invoke upgrade logic.